pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -103,28 +108,29 @@ func (q *QueryReconciler) collectInfoFromVdb(ctx context.Context) (ctrl.Result,
 	return res, err
 }
 
-// setListRestorePointsQueryConditions will update the status condition before and after calling
+// runListRestorePoints will update the status condition before and after calling
 // list restore points api
 // Temporarily, runListRestorePoints will not call the ListRestorePoints API
 // since the dispatcher is not set up yet
 func (q *QueryReconciler) runListRestorePoints(ctx context.Context, _ *ctrl.Request) error {
-	// set Querying status condition prior to calling vclusterops API
-	err := vrpqstatus.UpdateCondition(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
-		v1.MakeCondition(vapi.Querying, metav1.ConditionTrue, ""))
+	// set Querying status condition and state prior to calling vclusterops API
+	err := vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+		v1.MakeCondition(vapi.Querying, metav1.ConditionTrue, "Started"), querying)
 	if err != nil {
 		return err
 	}
 
 	// API should be called to proceed here
+	// If we receive a failure result from the API, a state message and condition need to be updated
 
 	// clear Querying status condition
-	err = vrpqstatus.UpdateCondition(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
-		v1.MakeCondition(vapi.Querying, metav1.ConditionFalse, ""))
+	err = vrpqstatus.UpdateConditionAndState(ctx, q.VRec.Client, q.VRec.Log, q.Vrpq,
+		v1.MakeCondition(vapi.Querying, metav1.ConditionFalse, "CompletedOrFailed"), querying)
```

Comment:
I decided to use the message 'CompletedOrFailed' when Go complained that 'Completed|Failed' does not match the regular expression pattern ^A-Za-z?$.`

Reply:
Sorry, I wasn't clear. I meant we would pick Completed or Failed, depending on the result of the action. Can you change it to Completed?